pytorch_to_returnn/torch/nn/utils/rnn.py

```diff
@@ -1,15 +1,31 @@
 import numpy
+from ..functional import full
 from ..modules.rnn import PackedSequence
 from ..modules.shape import FlattenBatch
 from ...tensor import Tensor
 from ..._C import from_numpy
+from .... import torch
 from ....naming import Naming
 
 
+def _batch_sizes_from_lengths(lengths: Tensor) -> Tensor:
+  indices = torch.arange(torch.max(lengths))  # [T]
```

Comment:
I think this does not work properly. `Range._get_output_shape_from_returnn` gets a `Tensor` for `limit`, but `limit.numpy()=0` because the `numpy_buffer` it not set correctly after `Reduce`.

Reply:
But then this sounds like yet another bug. This should work. Maybe you can make a separate test case for this.

Reply:
I pushed something to set the numpy buffer which fixes this here

Reply:
Maybe you can also make a separate test case specifically for sth like `torch.arange(torch.max(lengths))`.

Reply:
I added this test case (`test_arange_from_lengths`), it currently works.

Reply:
You say the test `test_arange_from_lengths` works? But so what's different here then? Maybe `lengths` is not constant. Where does this fail, in what test? Where does `lengths` come from?

Reply:
Oh sorry, I still have my numpy array modifications active which you wanted to remove.

Reply:
Hm no I didn't

Reply:
But this just does not fail because the torch shape is not checked. That's wrong `(0,)` here as well. In the other test case `test_packed_sequence_2` subsequently something fails because of the wrong torch shape I think.

Reply:
Then add some `assert arange.shape == ...` to the test `test_arange_from_lengths`.

Reply:
Yes, this makes the problem obvious. Just pushed it.